lib/engine/game/g_18_ms/game.rb

```diff
@@ -365,8 +365,10 @@ def revenue_for(route, stops)
         end
 
         def routes_revenue(routes)
+          return 0 if routes.empty?
+
           if @round.round_num == 2
-            active_step.current_entity.trains.each do |t|
+            routes.first.corporation.trains.each do |t|
```

Comment:
Changed dividend step to determine action on whether there is revenue or not, not on routes. This caused an infinite loop for 18MS since current_entity leads to actions being called, which in turn calls routes_revenue...Get the corporation from the route instead of current_entity to break the infinite loop.